packages/react-instantsearch-core/src/core/__tests__/createInstantSearchManager.result.js

```diff
@@ -1,323 +1,273 @@
-            const store = ism.store.getState();
-            expect(_dispatchAlgoliaResponse).toHaveBeenCalledTimes(2);
-            expect(store.results).toEqual({ value: 'results' });
-            expect(store.error).toBe(null);
-          });
+      const ism = createInstantSearchManager({
+        indexName: 'index',
+        searchClient,
       });
-    });
 
-    describe('on external updates', () => {
-      it('updates the store and searches', () => {
-        expect.assertions(7);
+      ism.widgetsManager.registerWidget({
+        getSearchParameters: params => params.setQuery('search'),
+        props: {},
+        context: {},
+      });
 
-        const _dispatchAlgoliaResponse = jest.fn(function(state) {
-          this.emit('result', { value: 'results' }, state);
-        });
+      expect(ism.store.getState().results).toBe(null);
 
-        jsHepler.mockImplementationOnce((...args) => {
-          const instance = jsHepler(...args);
+      return Promise.resolve()
+        .then()
+        .then(() => {
+          const store = ism.store.getState();
 
-          instance._dispatchAlgoliaResponse = _dispatchAlgoliaResponse;
+          expect(searchClient.search).toHaveBeenCalledTimes(1);
+          expect(store.results.hits).toEqual([{ value: 'results' }]);
+          expect(store.error).toBe(null);
 
-          return instance;
-        });
+          ism.widgetsManager.update();
+        })
+        .then()
+        .then(() => {
+          const store = ism.store.getState();
 
-        const ism = createInstantSearchManager({
-          indexName: 'index',
-          initialState: {},
-          searchParameters: {},
-          searchClient: client,
+          expect(searchClient.search).toHaveBeenCalledTimes(2);
+          expect(store.results.hits).toEqual([{ value: 'results' }]);
+          expect(store.error).toBe(null);
         });
-
-        ism.onExternalStateUpdate({});
-
-        expect(ism.store.getState().results).toBe(null);
-
-        return Promise.resolve()
-          .then(() => {
-            const store = ism.store.getState();
-            expect(_dispatchAlgoliaResponse).toHaveBeenCalledTimes(1);
-            expect(store.results).toEqual({ value: 'results' });
-            expect(store.error).toBe(null);
-
-            ism.widgetsManager.update();
-          })
-          .then(() => {})
-          .then(() => {
-            const store = ism.store.getState();
-            expect(_dispatchAlgoliaResponse).toHaveBeenCalledTimes(2);
-            expect(store.results).toEqual({ value: 'results' });
-            expect(store.error).toBe(null);
-          });
-      });
     });
 
-    describe('on search for facet values', () => {
-      it('updates the store and searches', () => {
-        expect.assertions(4);
-
-        const searchForFacetValues = jest.fn(() =>
-          Promise.resolve({
-            facetHits: 'results',
-          })
-        );
+    it('updates the store on external updates', () => {
```

Comment:
We only test that `onExternalStateUpdate` trigger a new search. The `nextSearchState` object provided in that case is useless because the registered widgets are not connected to the `searchState`.